test/consul_connect_tests.go

```diff
@@ -0,0 +1,137 @@
+package test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+	"github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/retry"
+	"github.com/gruntwork-io/terratest/modules/ssh"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	test_structure "github.com/gruntwork-io/terratest/modules/test-structure"
+)
+
+
+
+// Test the consul-cluster example by:
+//
+// 1. Copying the code in this repo to a temp folder so tests on the Terraform code can run in parallel without the
+//    state files overwriting each other.
+// 2. Building the AMI in the consul-ami example with the given build name
+// 3. Deploying that AMI using the consul-cluster Terraform code
+// 4. Checking that the Consul cluster comes up within a reasonable time period and can respond to requests
+func runConsulConnectTest(t *testing.T, packerBuildName string, examplesFolder string, packerTemplatePath string, sshUser string) {
+	runConsulConnectTestWithVars(t,
+		packerBuildName,
+		examplesFolder,
+		packerTemplatePath,
+		sshUser,
+		map[string]interface{}{})
+}
+
+func runConsulConnectTestWithVars(t *testing.T, packerBuildName string, examplesFolder string, packerTemplatePath string, sshUser string, terraformVarsMerge map[string]interface{}) {
```

Comment:
NIT: you might be working on this, but to avoid duplicating the code consider passing in a boolean param `checkConnect` which is used to enable connect tests.

Reply:
I thought about this and realized that for future testing we would need more tests on the service layers that may necessitate a separate helper for connect tests. Further, the overlap is much less now, but yes can be refactored some more